src/Dodo.HttpClient.ResiliencePolicies/RetrySettings/JitterRetrySettings.cs

```diff
@@ -8,45 +8,29 @@ namespace Dodo.HttpClientResiliencePolicies.RetrySettings
 {
 	public class JitterRetrySettings : IRetrySettings
 	{
-		public int RetryCount { get; }
-		public TimeSpan MedianFirstRetryDelay { get; }
-		public Func<int, TimeSpan> SleepDurationProvider { get; }
-		public Action<DelegateResult<HttpResponseMessage>, TimeSpan> OnRetry { get; set; }
-
-		public JitterRetrySettings(int retryCount) : this(retryCount, _defaultMedianFirstRetryDelay)
+		public JitterRetrySettings()
 		{
-		}
+			RetryCount = Defaults.Retry.RetryCount;
+			SleepDurationProvider = _defaultSleepDurationProvider(RetryCount, _defaultMedianFirstRetryDelay);
 
-		public JitterRetrySettings(int retryCount, Action<DelegateResult<HttpResponseMessage>, TimeSpan> onRetry) :
-			this(retryCount, _defaultMedianFirstRetryDelay, onRetry)
-		{
+			OnRetry = _doNothingOnRetry;
 		}
 
-		public JitterRetrySettings(int retryCount, TimeSpan medianFirstRetryDelay) : this(retryCount,
-			medianFirstRetryDelay, _defaultOnRetry)
-		{
-		}
+		public int RetryCount { get; set; }
 
-		public JitterRetrySettings(
-			int retryCount,
-			TimeSpan medianFirstRetryDelay,
-			Action<DelegateResult<HttpResponseMessage>, TimeSpan> onRetry)
-		{
-			RetryCount = retryCount;
-			SleepDurationProvider = _defaultSleepDurationProvider(retryCount, medianFirstRetryDelay);
-			OnRetry = onRetry;
-		}
+		public Func<int, TimeSpan> SleepDurationProvider { get; set; }
 
-		public static IRetrySettings Default() => new JitterRetrySettings(Defaults.Retry.RetryCount);
+		public Action<DelegateResult<HttpResponseMessage>, TimeSpan> OnRetry { get; set; }
 
 		private static readonly TimeSpan _defaultMedianFirstRetryDelay =
```

Comment:
Should we add MedianFirstRetryDelay as a public property too? It was possible to set it in the previous version of the JitterRetrySettings and it is important parameter for this type of settings.

Reply:
I think it doesn't make sense.

In the previous version, this parameter can be sent directly to sleepDurationProvider via constructor, and the property MedianFirstRetryDelay itself wasn't used.
`public JitterRetrySettings(
			int retryCount,
			TimeSpan medianFirstRetryDelay,
			Action<DelegateResult<HttpResponseMessage>, TimeSpan> onRetry)
		{
			RetryCount = retryCount;
			SleepDurationProvider = _defaultSleepDurationProvider(retryCount, medianFirstRetryDelay);
			OnRetry = onRetry;
		} `

The second thought is in this design version of settings classes If we add a public property MedianFirstRetryDelay we don't have control over when it will be filled and this means that we don't know when we should change the sleepDurationProvider.